File: php/browser/kelvinator_steps.rb
Given /^a temperature of (\d+) degrees centigrade$/ do |centigrade|
  @centigrade = centigrade.to_i
end

When /^I convert it to Kelvin$/ do
  browser.navigate.to "http://localhost/~#{ENV['USER']}/index.php" #<label id="code.php"/>
  input = browser.find_element :name, 'centigrade'
  input.send_keys @centigrade.to_s
  input.submit

  output = browser.find_element :id, 'kelvin'
  @kelvin = output.text.to_i
end

Then /^the result should be (\d+) degrees Kelvin$/ do |expected|
  @kelvin.should == expected.to_i
end

File: php/features/step_definitions/KelvinatorSteps.php
<?php
// START:require
require('kelvinator.php');
// END:require
?>

<?php /* START:structure */ ?>
<?php
class KelvinatorSteps extends CucumberSteps {

    // Your step definitions will go here

    /* END:structure */

    /* START:given */
    /**
    * Given /^a temperature of (\d+) degrees centigrade$/ <label id="code.php.regex"/>
    **/
    public function stepATemperatureOfDegreesCentigrade($centigrade) { // <label id="code.php.param"/>
        $this->aGlobals['centigrade'] = $centigrade; // <label id="code.php.array"/>
    }
    /* END:given */

    /* START:when */
    /**
     * When /^I convert it to Kelvin$/
     **/
    public function stepIConvertItToKelvin() {
        $this->aGlobals['kelvin'] = kelvinate(
            $this->aGlobals['centigrade']);
    }
    /* END:when */

    /* START:then */
    /**
     * Then /^the result should be (\d+) degrees Kelvin$/
     **/
    public function stepTheResultShouldBe3DegreesKelvin($expected) {
        self::assertEquals($this->aGlobals['kelvin'], $expected); // <label id="code.php.assert"/>
    }
    /* END:then */

    /* START:structure */
}
?>
<?php /* END:structure */ ?>
